service/matching/taskListManager.go

```diff
@@ -562,26 +562,35 @@ func (c *taskListManagerImpl) CancelPoller(pollerID string) {
 
 // Returns a batch of tasks from persistence starting form current read level.
 // Also return a number that can be used to update readLevel
-func (c *taskListManagerImpl) getTaskBatch() ([]*persistence.TaskInfo, int64, error) {
+func (c *taskListManagerImpl) getTaskBatch() ([]*persistence.TaskInfo, int64, bool, error) {
 	var tasks []*persistence.TaskInfo
 	readLevel := c.taskAckManager.getReadLevel()
```

Comment:
I think we need to make sure read level is initialized from last checkpointed ack level from task list metadata.

Reply:
Yes we are sure read level will be init with ack level when start